src/main/java/org/gridsuite/filter/server/dto/expertfilter/expertrule/NumberExpertRule.java

```diff
@@ -54,12 +64,18 @@ public boolean evaluateRule(Identifiable<?> identifiable) {
             case LOWER_OR_EQUALS -> identifiableValue.compareTo(filterValue) <= 0;
             case LOWER -> identifiableValue.compareTo(filterValue) < 0;
             case EXISTS -> true; // We return true here because we already test above if identifiableValue is NaN.
+            case IN -> this.getValues().contains(identifiableValue);
+            case NOT_IN -> !this.getValues().contains(identifiableValue);
             default -> throw new PowsyblException(this.getOperator() + " operator not supported with " + this.getDataType() + " rule data type");
         };
     }
 
     @Override
     public String getStringValue() {
-        return String.valueOf(this.getValue());
+        if (isMultipleCriteriaOperator(this.getOperator())) { // multiple values
+            return this.getValues().stream().map(String::valueOf).collect(Collectors.joining(","));
+        } else { // single value or absence
```

Comment:
Maybe return an empty string, `null` or `String.valueOf(Collections.<String>emptySet())` if `this.values` is null ?
This is to prevent a NPE.

Reply:
Should manage at DTO level by a validator to do a cross validation between two fields. This will be corrected in another PR